test/common/util.js

```diff
@@ -420,9 +439,82 @@ describe('common/util', function() {
         }
       });
     });
+
+    it('should emit an error if the request fails', function(done) {
+      var dup = duplexify();
+      var fakeStream = new stream.Writable();
+      var fakeResponse = {
+        statusCode: 400
+      };
+
+      fakeStream.write = function() {};
+      dup.end = function() {};
+
+      requestOverride = function() {
+        return fakeStream;
+      };
+
+      var options = {
+        makeAuthorizedRequest: function(request, opts) {
+          opts.onAuthorized();
+        }
+      };
+
+      util.makeWritableStream(dup, options);
+
+      dup.on('error', function(err) {
+        assert.strictEqual(err.code, fakeResponse.statusCode);
+        done();
+      });
+
+      setImmediate(function() {
+        fakeStream.emit('complete', fakeResponse);
```

Comment:
This will trigger this:

```js
stream.on('complete', function(res) {
  handleResp(null, res, res.body, function(err, data) {
    if (err) {
      dup.emit('error', err);
      dup.end();
      return;
    }
    onComplete(data);
  });
});
```

Your tests cover both of these cases well, but I would change the approach a little bit. Instead of letting `handleResp` actually run, make sure it's being called in the first place:

```diff
stream.on('complete', function(res) {
-  handleResp(null, res, res.body, function(err, data) {
+  util.handleResp(null, res, res.body, function(err, data) {
```

Then use `utilOverrides` to execute the callback with the data that will cause the error handling / success handling.